subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4268,6 +4270,9 @@ int lwm2m_parse_peerinfo(char *url, struct sockaddr *addr, bool *use_dtls)
 		hints.ai_family = AF_INET6;
 #elif defined(CONFIG_NET_IPV4)
 		hints.ai_family = AF_INET;
+#elif defined(CONFIG_NET_SOCKETS_OFFLOAD)
```

Comment:
IMHO this is conceptually wrong thing to do. Sockets offloading has nothing to do with DNS resolving so we should not have the check like this here. I think we need to think more about this. Is this fix really needed for 2.0?

Reply:
@jukkar Please consider the situation that neither CONFIG_NET_IPV4 nor CONFIG_NET_IPV6 are defined when CONFIG_NET_SOCKETS_OFFLOAD is used and the IP stack is not in Zephyr.

Reply:
@jukkar what if we change the flag to CONFIG_NET_OFFLOAD? It's the same level of configuration as CONFIG_NET_IPV4 and CONFIG_NET_IPV6, as seen in net_context.

Reply:
Changing to CONFIG_NET_OFFLOAD is no use here. The point is that we do here "magic" by checking CONFIG_NET_SOCKETS_OFFLOAD and then doing IPv4 DNS query. One option might be to create a new Kconfig option just for LWM2M, depend that on DNS and IPv4, and use that option here.

Reply:
@mike-scott Can we postpone this to 2.1?

Reply:
If this is really needed for 2.0, then we can do it like this. But this would need to be "fixed" properly to 2.1, perhaps #18153 helps here.

Reply:
@mike-scott I am fine with that.

Reply:
@mike-scott default handling in Nordic offload sockets implementation would work but it's just a workaround. Furthermore, AF_UNSPEC does mean something and we can't overwrite it with AF_INET in all cases. Hopefully we could use current commit and wait for new solution based on [#18153 ](https://github.com/zephyrproject-rtos/zephyr/pull/18153) after 2.1

Reply:
@junqingzou Ok, let's use this for now.  I will also have more modems using offloaded DNS in Zephyr 2.1 so we will have more information to examine then.